src/components/AntdPopconfirm/index.tsx

```diff
@@ -0,0 +1,72 @@
+import { useEffect, useState } from "react";
+import styled from "@emotion/styled";
+import Checkbox from "@leafygreen-ui/checkbox";
+import { Popconfirm as AntPopconfirm } from "antd";
+import { size } from "constants/tokens";
+
+interface Props extends React.ComponentProps<typeof AntPopconfirm> {
+  children: React.ReactNode;
+}
+
+export const Popconfirm: React.VFC<Props> = ({ children, ...props }) => (
+  // eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions
+  <span
+    onClick={(e) => {
+      e.stopPropagation(); // Stop click propagation for clicks on Popconfirm popup body.
+    }}
+  >
+    <AntPopconfirm {...props}>{children}</AntPopconfirm>
+  </span>
+);
+
+interface PopconfirmWithCheckboxProps {
+  onConfirm: (e: React.MouseEvent) => void;
+  title: string;
+  checkboxLabel?: string;
+  children: React.ReactNode;
+}
+
+export const PopconfirmWithCheckbox: React.VFC<PopconfirmWithCheckboxProps> = ({
+  checkboxLabel, // truthiness determines if checkbox is rendered
```

Comment:
My above comment aside assuming you don't want to make those changes in this pr. 

I'm not sure how I feel about this implementation. If a user omits a checkbox label then this component is effectively just a normal Popconfirm. Either we should bake this functionality into the original Popconfirm or we should make checkboxLabel a required prop.

Reply:
This is not actually my change (I just renamed the file), but sure I can make it a required prop!